### ISSUE TITLE
Clarify Connection pooling section

### ISSUE THREAD

**Issue by jvmlet** [MicrosoftDocs/sql-docs#8753]
[This ](https://learn.microsoft.com/en-us/sql/relational-databases/server-management-objects-smo/create-program/connecting-to-an-instance-of-sql-server?view=azure-sqldw-latest#connection-pooling
) section is **VERY** misleading 

#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: b51309c7-1370-863d-4a44-eefe2d381161
* Version Independent ID: e56a7f00-aa5a-c0e7-3862-77b3e85f3a65
* Content: [Connecting to an Instance of SQL Server - SQL Server](https://learn.microsoft.com/en-us/sql/relational-databases/server-management-objects-smo/create-program/connecting-to-an-instance-of-sql-server?view=azure-sqldw-latest#connection-pooling)
* Content Source: [docs/relational-databases/server-management-objects-smo/create-program/connecting-to-an-instance-of-sql-server.md](https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/relational-databases/server-management-objects-smo/create-program/connecting-to-an-instance-of-sql-server.md)
* Service: **sql**
* GitHub Login: @markingmyname
* Microsoft Alias: **maghan**

**Comment by jvmlet** [comment]
> When the [Connect](https://learn.microsoft.com/en-us/dotnet/api/microsoft.sqlserver.management.common.connectionmanager.connect) method is called, the connection is not released to the pool

I wouldn't expect connection to be released to the pool when calling `Connect`

and 
> SMO will automatically establish a connection when required, and release the connection to the connection pool after it has finished performing operations

goes against

>An explicit call to the [Disconnect](https://learn.microsoft.com/en-us/dotnet/api/microsoft.sqlserver.management.common.connectionmanager.disconnect) method is required to release the connection to the pool.

**Comment by WilliamAntonRohm** [comment]
@jvmlet -- Furer, thank you for your follow-up.

@markingmyname -- Mark, please look into this issue.

**Comment by markingmyname** [comment]
@shueybubbles 

Any thoughts?

**Comment by shueybubbles** [comment]
The intent is that if the application calls `Connect` directly, SMO will not automatically disconnect when it's done. IE, the lifetime of the connection changes to be explicitly controlled by the caller.

```
When the [Connect](https://learn.microsoft.com/en-us/dotnet/api/microsoft.sqlserver.management.common.connectionmanager.connect) method is called, the connection is not released to the pool. An explicit call to the [Disconnect](https://learn.microsoft.com/en-us/dotnet/api/microsoft.sqlserver.management.common.connectionmanager.disconnect) method is required to release the connection to the pool.
```

**Comment by jvmlet** [comment]
Well, I do admit English is not my mother tongue, but IMO even native English speaker has difficulty to decrypt the complexity both of the sentence and logic behind it.
Explicitly calling `Connect()` changes the ownership of the connection ???

**Comment by shueybubbles** [comment]
I agree the language needs to be updated.
What I mean by 'lifetime of the connection' is something like this. We should probably write a unit test to both verify this behavior and document it more clearly.
*** I'm writing this without verifying the behavior but I can try to get to it at some point. ***

Without a `Connect` call by the app:

```C#
var serverConnection = new ServerConnection(sqlConnection);
var smoServer = new Server(serverConnection);
var dbName = smoServer.Databases["mydb"].Name;
Assert.That(sqlConnection.State, Is.EqualTo(ConnectionState.Closed))
```

With a `Connect` call:
```C#
var serverConnection = new ServerConnection(sqlConnection);
serverConnection.Connect();
var smoServer = new Server(serverConnection);
var dbName = smoServer.Databases["mydb"].Name;
Assert.That(sqlConnection.State, Is.EqualTo(ConnectionState.Open))
serverConnection.Disconnect();
Assert.That(sqlConnection.State, Is.EqualTo(ConnectionState.Closed))
```

**Comment by jvmlet** [comment]
@shueybubbles , I suggest to continue the discussion [here](https://github.com/microsoft/sqlmanagementobjects/issues/143), but from what I understood, if the caller invoked `Connect`  - he **HAS** to call `Disconnect` as well. 
I'm OK  with this  approach, just wanted to make sure there are no any unpredicted/non-straightforward/non-logical   **_side-effects_** from API  perspective.

**Comment by markingmyname** [comment]
I will reword it to say something more like this:


Calling the Connect method of the ServerConnection object is generally unnecessary. After operations are completed, SMO will automatically establish connections when needed and return them to the connection pool. If you call the Connect method, the connection won't be released to the pool. To achieve that, you need to use the Disconnect method explicitly. Furthermore, you can acquire a non-pooled connection by adjusting the NonPooledConnection property of the ServerConnection object.

**Comment by markingmyname** [comment]
Updates will be applied end of the week.

Thank you, @shueybubbles, for assisting.

**Comment by markingmyname** [comment]
We will now proceed to close this thread. If there are further questions regarding this matter, please tag me in your reply. We will gladly continue the discussion, and we will reopen the issue. To see what the community is wanting, please visit the [SQL Server User Feedback](https://feedback.azure.com/d365community/forum/04fe6ee0-3b25-ec11-b6e6-000d3a4f0da0) site or the [Q & A](https://docs.microsoft.com/en-us/answers/products/sql-server) site.
